[Swift] Support WebKitAdditions in swift-args.resp and work around dependency bug

emw-apple opened the pull request:
#### aac8c557da9ae65151ba533ba8d5b1c7841e741c
<pre>
[Swift] Support WebKitAdditions in swift-args.resp and work around dependency bug
<a href="https://bugs.webkit.org/show_bug.cgi?id=292241">https://bugs.webkit.org/show_bug.cgi?id=292241</a>
<a href="https://rdar.apple.com/147084175">rdar://147084175</a>

Reviewed by Alexey Proskuryakov and Richard Robinson.

Instead of running `generate-platform-args` from WTF, during its
installhdrs phase, have every project that compiles Swift sources invoke
it locally. Where appropriate, this script is added to projects&apos;
&quot;Derived Sources&quot; target, otherwise it&apos;s just run before that target
begins compiling.

We recently discovered that Swift doesn&apos;t emit input dependencies for
@-style response files (<a href="https://rdar.apple.com/150228472">rdar://150228472</a>), so WebKit&apos;s incremental build
is broken when Platform.h changes. AFAICT, there&apos;s not an easy way in
Xcode to incorporate the .resp file into a target&apos;s &quot;CompileSwift&quot; task.
Instead, as a hacky workaround, have `generate-platform-args` emit an
empty .swift file as a side effect. Have each target that compiles Swift
code compile this file. Then, when platform args are regenerated, this
file will be touched, Xcode will invoke the Swift driver, and Swift will
realize it now has different -D arguments and recompile code as needed.

Lastly, as a drive-by, accept changes from `sort-Xcode-project-file` to
the projects this commit touches.

* Configurations/CommonBase.xcconfig:
* Source/WTF/Scripts/generate-platform-args:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/Platform.h:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Aside from the main
  change, also remove two files from &quot;Product Dependencies&quot; that don&apos;t
  belong (they are not build products).
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPIBundle.xcconfig:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/294347@main">https://commits.webkit.org/294347@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3f8edddc4c4e7ee0fff9118e9f65e3008abc2917

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/101404 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/21069 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/11374 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/106557 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/52033 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/103444 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/21376 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/29562 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/77243 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/34268 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/104411 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/16504 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/91592 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/57585 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/16325 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/9607 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/51381 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/94073 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/86201 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/9680 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/108909 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/100014 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/28533 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/20993 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/86221 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/28895 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/87796 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/85783 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21844 "Built successfully and passed tests") | [❌ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/30503 "Found 1 new test failure: resize-observer/contain-instrinsic-size-should-not-leak-nodes.html (failure)") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/8213 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/22650 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/28464 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/33744 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/123639 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/28275 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/34382 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/31595 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/29834 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
It's not clear to me how having one empty file helps Swift realize that all other files need to be recompiled.

emw-apple committed:
[Swift] Support WebKitAdditions in swift-args.resp and work around dependency bug
https://bugs.webkit.org/show_bug.cgi?id=292241
rdar://147084175

Reviewed by Alexey Proskuryakov and Richard Robinson.

Instead of running `generate-platform-args` from WTF, during its
installhdrs phase, have every project that compiles Swift sources invoke
it locally. Where appropriate, this script is added to projects'
"Derived Sources" target, otherwise it's just run before that target
begins compiling.

We recently discovered that Swift doesn't emit input dependencies for
@-style response files (rdar://150228472), so WebKit's incremental build
is broken when Platform.h changes. AFAICT, there's not an easy way in
Xcode to incorporate the .resp file into a target's "CompileSwift" task.
Instead, as a hacky workaround, have `generate-platform-args` emit an
empty .swift file as a side effect. Have each target that compiles Swift
code compile this file. Then, when platform args are regenerated, this
file will be touched, Xcode will invoke the Swift driver, and Swift will
realize it now has different -D arguments and recompile code as needed.

Lastly, as a drive-by, accept changes from `sort-Xcode-project-file` to
the projects this commit touches.

* Configurations/CommonBase.xcconfig:
* Source/WTF/Scripts/generate-platform-args:
* Source/WTF/WTF.xcodeproj/project.pbxproj:
* Source/WTF/wtf/Platform.h:
* Source/WTF/wtf/PlatformEnable.h:
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/PAL.xcodeproj/project.pbxproj:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Aside from the main
  change, also remove two files from "Product Dependencies" that don't
  belong (they are not build products).
* Source/WebGPU/Configurations/WebGPU.xcconfig:
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebKit/Configurations/Base.xcconfig:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Configurations/TestWebKitAPIBundle.xcconfig:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/294347@main
